plugins/statuslines/lualine.nix

```diff
@@ -1,260 +1,281 @@
-                  icon = mkOption {
-                    type = str;
-                    description = "Icon character.";
-                  };
-                };
-              })
-            ) "Defines the icon to be displayed in front of the component.";
-
-            separator = mkSeparatorsOption { };
-
-            color = helpers.mkNullOrOption (types.attrsOf types.str) ''
-              Defines a custom color for the component.
-            '';
-
-            padding = helpers.defaultNullOpts.mkNullable (types.either types.int (
-              types.submodule {
                 options = {
                   left = mkOption {
-                    type = types.int;
-                    description = "left padding";
+                    type = int;
+                    description = "Left padding.";
                   };
                   right = mkOption {
-                    type = types.int;
-                    description = "left padding";
+                    type = int;
+                    description = "Right padding.";
                   };
                 };
-              }
-            )) 1 "Adds padding to the left and right of components.";
-
-            fmt = helpers.mkNullOrLuaFn ''
-              A lua function to format the component string.
-
-              Example:
-              ```lua
-                function(text)
-                  return text .. "!!!"
-                end
-              ```
-            '';
-
-            extraConfig = mkOption {
-              type = types.attrs;
-              default = { };
-              description = "extra options for the component";
+              })) 1 "Adds padding to the left and right of components.";
+
+              fmt = lib.nixvim.mkNullOrLuaFn' {
+                description = ''
+                  A lua function to format the component string.
+                '';
+                example = ''
+                  ```lua
+                    function(text)
+                      return text .. "!!!"
+                    end
+                  ```
+                '';
+              };
             };
           };
-        })
-      )
-    ) "";
+        in
+        listOf (lib.nixvim.transitionType oldType coerceFn newType);
+      description = "";
+    };
 
   mkEmptySectionOption = name: {
-    lualine_a = mkComponentOptions "";
-    lualine_b = mkComponentOptions "";
-    lualine_c = mkComponentOptions "";
-    lualine_x = mkComponentOptions "";
-    lualine_y = mkComponentOptions "";
-    lualine_z = mkComponentOptions "";
+    lualine_a = mkComponentOptions "Left section on left side.";
+    lualine_b = mkComponentOptions "Middle section on left side.";
+    lualine_c = mkComponentOptions "Right section on left side.";
+    lualine_x = mkComponentOptions "Left section on right side.";
+    lualine_y = mkComponentOptions "Middle section on right side.";
+    lualine_z = mkComponentOptions "Right section on right side.";
   };
 in
```

Comment:
Should we move `mkSeparatorsOption`, `mkComponentOptions`, and `mkEmptySectionOption` to a let-block local to `settingsOptions`?

Might make the top section of the file a little more readable?